src/test/java/com/google/sps/DatastorePlaceGuideRepositoryTest.java

```diff
@@ -0,0 +1,303 @@
+package com.google.sps;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.RepositoryType;
+import com.google.sps.placeGuide.PlaceGuide;
+import com.google.sps.placeGuide.repository.PlaceGuideRepository;
+import com.google.sps.placeGuide.repository.PlaceGuideRepositoryFactory;
+import com.google.sps.placeGuide.repository.impl.DatastorePlaceGuideRepository;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(JUnit4.class)
+public final class DatastorePlaceGuideRepositoryTest {
+  
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(
+                                                            new LocalDatastoreServiceTestConfig());
+
+  public static final long A_PUBLIC_ID = 12345;
+  public static final long B_PUBLIC_ID = 23456;
+  public static final long A_PRIVATE_ID = 34567;
+  public static final long B_PRIVATE_ID = 45678;
+  public static final String NAME = "name";
+  public static final String AUDIO_KEY = "audioKey";
+  public static final String CREATOR_A_ID = "creatorA_Id";
+  public static final String CREATOR_B_ID = "creatorB_Id";
+  public static final String OTHER_USER_ID = "otherUserId";
+  public static final String PLACE_ID = "placeId";
+  public static final GeoPt COORDINATE = new GeoPt((float) 3.14, (float) 2.56);
+  public static final boolean IS_PUBLIC = true;
+  public static final long LENGTH = new Long(60);
+  public static final String DESCRIPTION = "description";
+  public static final String PREVIOUS_DESCRIPTION = "previous description";
+  public static final String IMAGE_KEY = "imageKey";
+
+  private final PlaceGuide testPublicPlaceGuideA = 
+                                            new PlaceGuide.Builder(A_PUBLIC_ID, NAME, AUDIO_KEY, 
```

Comment:
too many spaces, make it 4. Here and below

The standard is when continue from previous line, add 4 spaces at the beginning.

Reply:
fixed spaces.